consoles/console.pm

```diff
@@ -63,9 +63,17 @@ sub screen {
 
 # helper function
 sub sshCommand {
-    my ($self, $username, $host, $gui) = @_;
+    my ($self, $username, $host, $gui, $key) = @_;
 
-    my $sshopts = "-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PubkeyAuthentication=no $username\@$host";
+    my $sshopts = "-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no";
+
+    if ($key) {
+        $sshopts = $sshopts . " -i $key";
+    }
+    else {
+        $sshopts .= " -o PubkeyAuthentication=no";
+    }
```

Comment:
Please see my further comments in https://github.com/os-autoinst/os-autoinst/pull/1038#pullrequestreview-164239585